.github/workflows/test-build.yml

```
@@ -119,9 +137,36 @@ jobs:
           # See https://github.com/microsoft/vscode-ripgrep#github-api-limit-note
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Print information about build
+      - uses: actions/checkout@v4
+        if: ${{ ! vars.SKIP_BUILD && github.event.inputs.rari-ref != '' }}
```

Nit: These `!a && b != ""` conditions confused me a bit. How about:
```suggestion
        if: ${{ ! vars.SKIP_BUILD || github.event.inputs.rari-ref == '' }}
```

---

I guess you missed some brackets and mean `!( vars.SKIP_BUILD || github.event.input.rari-ref == '' )` I tend to find NNF more understandable, but in this case I agree a big wrapping negation makes sense and turns the `if` into an `unless`